### DISCUSSION TITLE
Autogluon on very small dataset - Overfitting + Weighted ensemble always performs worse than individual classifiers

### DISCUSSION THREAD

**Discussion by apavlo89** [autogluon/autogluon#3392] [Q&A]
I am working on a small binary classification dataset of about 2000 samples so roughly 1700 are in the training set. I am currently toying with feature selection algos but my features range from 50 to 150. My best balanced accuracy can range from 65% to 60% on the training set but i do much worse on the test set (from 53% to 58%). Moreover, what is consistent is that the WeightedEnsemble model is never the best model. There is clear overfitting going on overall and I was wondering what can I do to fix it? Do you guys have any suggestions in regard to modifying Autogluon in a way that prevents overfitting on small datasets? I am eagerly awaiting for Autogluon 9.0 as I've seen that finally nested cv will be included. This addition may fix my issue once and for all, however is there anything I can do now? Anyone from the dev team has the beta nested cv code i could use? :P Any other suggestions on what I could do in the meantime???

P.S The algo I am training is to predict UFC winners. If anyone posts a good suggestion to fix my overfitting issue, I will post ufc winner predictions for fight night this Saturday :P

**Comment by apavlo89** [comment]
From what I am seeing, I probably need to create custom hyperparameter models from the ones that Autogluon uses (XT, LightGBM, CatBoost, XGBoost, Random Forest, KNN that have hyperparameters that try to avoid overfitting (like max_depth being< 5). Any idea how I can set these custom hyperparameters? The tutorials have me confused :/

As you can see I am currently using pretty much the default settings of Autogluon after applying feature selection:
```

import pandas as pd
from sklearn.model_selection import train_test_split
from imblearn.combine import SMOTEENN
from autogluon.tabular import TabularPredictor

# Read the top features from the CSV file
top_features = pd.read_csv('/content/drive/MyDrive/Colab Notebooks/top_features.csv')
top_feature_list = top_features.values.flatten().tolist()[1:] # Exclude index at position 0

## Filter the features in X_train and X_test to keep only the top features
X_train_important = X_train[top_feature_list]
X_test_important = X_test[top_feature_list]

# Combine X and y to create a data frame for AutoGluon
train_data_important = pd.concat([X_train_important, y_train], axis=1)
test_data_important = pd.concat([X_test_important, y_test], axis=1)

# Combine X and y to create a data frame for AutoGluon
#train_data_important = pd.concat([X_train, y_train], axis=1)
#test_data_important = pd.concat([X_test, y_test], axis=1)


######################################################################################################################
######################################################################################################################
metric= 'balanced_accuracy'
# Set training time limit
time_limit_s=3600*23

# Define the AutoGluon task and train the model
predictor = TabularPredictor(label='WINNER',eval_metric=metric, path=output_folder + '/EXPERIMENTAL7_WINNERPRED_maxd0point4_autogluon_balancedaccuracy_2010-latest_upcomingfight_08072023_fightnightintraindata_train-testtimesplit_v1').fit(train_data=train_data_important, presets='best_quality', time_limit=time_limit_s,calibrate_decision_threshold=True)

# Prediction
predictions = predictor.predict(test_data_important)
predictor.evaluate(test_data_important, silent=True)

# Finally, the leaderboard
leaderboard = predictor.leaderboard(test_data_important, extra_info=True)
print(leaderboard)
```
